cmake/modules/FindPETSC.cmake

```diff
@@ -197,16 +197,28 @@ IF(NOT PETSC_PETSCVARIABLES MATCHES "-NOTFOUND")
   ENDFOREACH()
 ENDIF()
 
+IF(PETSC_WITH_MPIUNI)
+  #
+  # Workaround: Some distributions happen to not install petscvariables and
+  # we consequently might miss some essential include directories. Let's
+  # try at least to find the mpiuni include directory.
+  #
+  DEAL_II_FIND_PATH(PETSC_INCLUDE_DIR_MPIUNI mpi.h
+  HINTS ${PETSC_INCLUDE_DIR_COMMON} ${PETSC_INCLUDE_DIR_ARCH} ${_petsc_includes}
+  PATH_SUFFIXES mpiuni petsc/mpiuni
```

Comment:
Very little. The hints take precedence. We only running into that risk if `${PETSC_INCLUDE_DIR_COMMON}` ever happens to be `/usr/include`. But it should always end in `.../petsc`. (Well as long as no one mutilated the petsc installation...)

I can make this check even more strict if you want.